osquery/tables/system/darwin/authdb.mm

```diff
@@ -0,0 +1,99 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <osquery/core.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+
+#include "osquery/tables/system/darwin/authdb.h"
+
+namespace osquery {
+namespace tables {
+
+QueryData genOSXAuthDB(QueryContext& context) {
+  
+  QueryData results;
+  CFArrayRef app_map = nullptr;
+  Row r;
+
+  if (context.constraints["domain"].exists(EQUALS)) {
+    // If a specific domain is requested, speed up the set of type conversions.
+    auto domains = context.constraints["domain"].getAll(EQUALS);
+    app_map = (CFArrayRef)CFArrayCreateMutable(
+        kCFAllocatorDefault, domains.size(), &kCFTypeArrayCallBacks);
+    for (const auto& domain : domains) {
+      auto cf_domain = CFStringCreateWithCString(kCFAllocatorDefault,
+                                                 domain.c_str(),
+                                                 kCFStringEncodingASCII);
+      CFArrayAppendValue((CFMutableArrayRef)app_map, cf_domain);
+      CFRelease(cf_domain);
+    }
+  } 
+
+  if (app_map == nullptr) {
```

Comment:
Not sure how to get AuthorizationRightGet() to wildcard search. I'll investigate more next week.